firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -228,6 +228,17 @@ public InstallationTokenResult generateAuthToken(
         if (httpResponseCode == 200) {
           return readGenerateAuthTokenResponse(httpsURLConnection);
         }
+
+        if (httpResponseCode == 401) {
+          return TokenResult.builder()
+              .setResponseCode(TokenResult.ResponseCode.REFRESH_TOKEN_ERROR)
+              .build();
+        }
+
+        if (httpResponseCode == 404) {
+          return TokenResult.builder().setResponseCode(TokenResult.ResponseCode.FID_ERROR).build();
+        }
```

Comment:
question: why does this not trigger a deletion of the PersistedInstallationEntry?
and if that is done somewhere else, could you please add a comment here, that the FID_ERROR will trigger a reset of the FIS SDK elsewhere?